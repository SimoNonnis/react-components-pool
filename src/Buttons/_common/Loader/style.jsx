import styled from 'styled-components';
import Spinner from '../../../Icons/Spinner';

export const SpinnerContainer = styled.div`
  position: relative;
`;

export const SpinnerStyled = styled(Spinner)`
  position: absolute;
  margin: 0 auto;
  left: 0;
  right: 0;
  top: -3px;
  ${({ size }) => size === 'large' && 'top: -5px'};
  ${({ size }) => size === 'small' && 'top: 0'};
`;
