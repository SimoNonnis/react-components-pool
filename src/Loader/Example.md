Loader component with defailt props
```jsx
<div style={{height: '100px', position: 'relative'}}>
  <Loader />
</div>
```
Loader component with all custom props
```jsx
<div style={{ height: '200px', position: 'relative'}}>
  <Loader background="pink" opacity="0.2" color="red" iconWidth={20} iconHeight={20} width="150px" height="150px" />
</div>
```