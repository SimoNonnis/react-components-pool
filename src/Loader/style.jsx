import styled from 'styled-components';

export const Container = styled.div`
  z-index: 99997;
  position: absolute;
  ${({ height, width }) => `
    height: ${height};
    width: ${width};
  `};
  &:before {
    content: '';
    position: absolute;
    z-index: 99998;
    width: 100%;
    height: 100%;
    left: 0;
    top: 0;
    ${({ background, opacity }) => `
      background-color: ${background};
      opacity: ${opacity};
    `};
  }
`;

export const SpinnerWrap = styled.div`
  left: 0;
  margin: 0 auto;
  position: relative;
  right: 0;
  text-align: center;
  top: 50%;
  transform: translateY(-50%);
  z-index: 99999;
`;
