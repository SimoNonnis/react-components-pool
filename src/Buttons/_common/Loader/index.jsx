import React from 'react';
import PropTypes from 'prop-types';

import { SpinnerStyled, SpinnerContainer } from './style';

const propTypes = {
  size: PropTypes.oneOf(['small', 'large', '']).isRequired,
};

const iconSize = {
  small: 25,
  large: 35,
  '': 30,
};

const Loader = ({ size }) => (
  <SpinnerContainer>
    &nbsp;
    <SpinnerStyled size={size} width={iconSize[size]} height={iconSize[size]} />
  </SpinnerContainer>
);

Loader.propTypes = propTypes;

export default Loader;
