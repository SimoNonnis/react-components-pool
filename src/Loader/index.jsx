import React from 'react';
import PropTypes from 'prop-types';

import theme from 'themes';

import { Spinner } from '../Icons';
import { Container, SpinnerWrap } from './style';

const propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  iconWidth: PropTypes.number,
  iconHeight: PropTypes.number,
  width: PropTypes.string,
  height: PropTypes.string,
  background: PropTypes.string,
  opacity: PropTypes.string,
};
const defaultProps = {
  className: null,
  color: theme.color.mediumBlue,
  iconWidth: 40,
  iconHeight: 40,
  width: '100%',
  height: '100%',
  background: '#ffffff',
  opacity: '0.7',
};

const Loader = ({
  className,
  color,
  iconWidth,
  iconHeight,
  width,
  height,
  background,
  opacity,
}) => (
  <Container
    id="loader"
    className={className}
    width={width}
    height={height}
    background={background}
    opacity={opacity}
  >
    <SpinnerWrap>
      <Spinner fill={color} width={iconWidth} height={iconHeight} />
    </SpinnerWrap>
  </Container>
);

Loader.propTypes = propTypes;
Loader.defaultProps = defaultProps;

export default Loader;
