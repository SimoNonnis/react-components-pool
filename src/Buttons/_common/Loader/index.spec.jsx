/* eslint-env jest */
import React from 'react';

import Loader from './';

describe('Test <Loader />', () => {
  it('should render Loader correctly with large props', () => {
    const wrapper = shallow(<Loader size="large" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Loader correctly with small props', () => {
    const wrapper = shallow(<Loader size="small" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render Loader correctly with empty size props', () => {
    const wrapper = shallow(<Loader size="" />);
    expect(wrapper).toMatchSnapshot();
  });
});
