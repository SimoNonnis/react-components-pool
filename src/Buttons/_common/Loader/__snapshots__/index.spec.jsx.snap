// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Loader /> should render Loader correctly with empty size props 1`] = `
<styled.div>
   
  <Styled(Spinner)
    height={30}
    size=""
    width={30}
  />
</styled.div>
`;

exports[`Test <Loader /> should render Loader correctly with large props 1`] = `
<styled.div>
   
  <Styled(Spinner)
    height={35}
    size="large"
    width={35}
  />
</styled.div>
`;

exports[`Test <Loader /> should render Loader correctly with small props 1`] = `
<styled.div>
   
  <Styled(Spinner)
    height={25}
    size="small"
    width={25}
  />
</styled.div>
`;
